examples/error_callback.rs

```diff
@@ -0,0 +1,34 @@
+use cpal::traits::HostTrait;
+use rodio::source::SineWave;
+use rodio::Source;
+use std::error::Error;
+use std::thread;
+use std::time::Duration;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    // You can use any other output device that can be queried from CPAL.
+    let default_device = cpal::default_host()
+        .default_output_device()
+        .ok_or("No default audio output device is found.")?;
+
+    let stream_handle = rodio::OutputStreamBuilder::from_device(default_device)?
+        .with_error_callback(|err| {
+            // Filter for where err is a DeviceNotAvailable error.
```

Comment:
most users will want to use the callback with an mpsc::Sender right? I might make sense to have a little more complex example that illustrates that.

Maybe the callback could send `cpal::StreamError::DeviceNotAvailable` into a std::mpsc and print all the others?